vizro-core/tests/conftest.py

```diff
@@ -1,10 +1,19 @@
 import pytest
 
-from vizro.managers import data_manager, model_manager
+from vizro import Vizro
 
 
 @pytest.fixture(autouse=True)
 def reset_managers():
+    # this ensures that the managers are reset before and after each test
+    # in principle clearing after every test should suffice, but this serves as a safety measure
+    # there are two tests in `test_get_action_loop_components.py` that fail without this
+    Vizro._clear_state()
```

Comment:
Do we know why we need the clearing twice?

Reply:
This is the strange issue I was telling you about - for some reason without the buildup clear there is some remnant left that affects the above two mentioned tests. So I am not entirely sure... Clearing in the build-up and teardown is also generally a more safe way to handle, so I think this is in principle good.

Reply:
I just tried with `hatch run test-unit` and the `reset_managers` implementation: 
1. Vizro._clear_state(); yield
2. yield; Vizro._clear_state()
and in both cases I get all tests passed. Did I overlook something?

Reply:
No this may work, as it executes the code before all tests. I suppose it is then for added safety to also execute after very test.

Reply:
It is even more complicated than that...

When running all tests the above is fine, just not when running this test alone. Going back to this I discovered that this runs fine if you `_reset` BEFORE the test, but not after.

It seems that when running this with other tests, the teardown of a previous test fixes the issue, taking the role of `_reset` before manually.

This feel like that there is something that it interferes with that needs to be explicitly be cleared. The error is quite strange, it starts to complain all of a sudden for the `Accordion` validation:

```
E   pydantic.error_wrappers.ValidationError: 1 validation error for Accordion
E   pages -> SELECT PAGE
E     none is not an allowed value (type=type_error.none.not_allowed)
```

Reply:
This was ressolved on Friday - the culprit was/is instatiating models outside of fixtures, in this case the harmful model was `vm.Navigation` in the navigation tests.

However since clearing the managers before every test is valid solution to this, and also because it can be quite convenient to define models outside fixtures (e.g. in parametrization) we decided that this was the correct course of action